manual/sphinx/user_docs/petsc.rst

```diff
@@ -0,0 +1,40 @@
+.. default-role:: math
+
+.. _sec-petsc:
+
+
+PETSc solvers
+=============
+
+Options for PETSc solvers can be passed in the input file (or on the command line).
+Global options are set in the ``[petsc]`` section. To set options specific to a
+particular PETSc-based solver, the options can be set in a ``petsc`` subsection of the
+solver's options, e.g. for a LaplaceXY solver (using the default options section) use the
+``[laplacexy:petsc]`` section. Note that the global options, including any
+passed on the command line [*]_, will be ignored for that solver if the subsection
+is created.
+
+Any options that can be passed on the command line to PETSc can be set, with no preceding
+hyphen. Any PETSc options that are passed as flags with no value set should be given the
+value ``true`` so BOUT++ knows to read them. So for example, if the command line options
+would be::
+
+    -ksp_monitor -ksp_type gmres
+
+in the input file you would put::
+
+    [petsc]
+    ksp_monitor = true
+    ksp_type = gmres
+
+
+.. [*] The object-specific options are passed to PETSc by creating an object-specific
+       prefix ``boutpetsclib#_``, where ``#`` is replaced with an integer counter,
```

Comment:
Ah I see -- this sounds like it might be tricky to be sure of. Could we provide a way to set the specific prefix to use for a particular instance of something in the object constructor?

Reply:
I wondered about that, but wasn't sure if anyone would actually use it... I think it should be straightforward to do, we can add an optional argument to the constructor, so the user can set a name.

Reply:
I don't think we need to do anything necessary here, we should just be able to use the usual BOUT++ mechanisms.

That is, instead of doing

```
./model -petsc_option
```

we can now do

```
./model laplacexy:petsc:petsc_option=value
```

More verbose, sure, but also more specific.

Reply:
This footnote can go now

Reply:
:+1: good point. I'm testing out the current version now. Will update the manual before calling this PR ready to merge.